src/explorer/siteNodeBase.ts

```diff
@@ -197,46 +188,42 @@ export class SiteNodeBase extends NodeBase {
         // the scm url lives in the 1 index of enabledHostNames, not 0
         const remote = `https://${username}:${password}@${repo}`;
 
-
-        let git = require('simple-git/promise')(workspace.rootPath);
-
+        const git = require('simple-git/promise')(workspace.rootPath);
         try {
-            let status = await git.status();
+
+            const status = await git.status();
             if (status.files.length > 0) {
                 window.showWarningMessage(`There ${status.files.length > 1 ? 'are' : 'is'} ${status.files.length} uncommitted change${status.files.length > 1 ? 's' : ''} in local repo "${workspace.rootPath}"`);
             }
-            await git.push(remote, 'master');
-        }
-        catch (err) {
+            await git.push(remote, 'HEAD:master');
+        } catch (err) {
             if (err.message.indexOf('spawn git ENOENT') >= 0) {
-                let input = await window.showErrorMessage(`Git must be installed to use Local Git Deploy.`, `Install`)
+                const input = await window.showErrorMessage(`Git must be installed to use Local Git Deploy.`, `Install`);
                 if (input === 'Install') {
                     opn(`https://git-scm.com/downloads`);
                 }
                 throw err;
             } else if (err.message.indexOf('error: failed to push') >= 0) {
-                let input = await window.showErrorMessage(`Push rejected due to Git history diverging. Force push?`, `Yes`)
+                const input = await window.showErrorMessage(`Push rejected due to Git history diverging. Force push?`, `Yes`);
                 if (input === 'Yes') {
-                    await git.push(['-f', remote, 'master']);
+                    await git.push(['-f', remote, 'HEAD:master']);
                 } else {
                     throw err;
                 }
-            } else {
-                throw err;
             }
         }
 
         const newDeployment = !this._isSlot ?
             await this.webSiteClient.webApps.listDeployments(this.site.resourceGroup, this._siteName) :
-            await this.webSiteClient.webApps.listDeploymentsSlot(this.site.resourceGroup, this._siteName, this._slotName);
+            await this.webSiteClient.webApps.listDeploymentsSlot(this.site.resourceGroup, this._siteName, util.extractDeploymentSlotName(this.site));
         // if the oldDeployment has a length of 0, then there has never been a deployment
         if (newDeployment.length && oldDeployment.length &&
             newDeployment[0].deploymentId === oldDeployment[0].deploymentId) {
             await window.showErrorMessage(`Local Git repo is current with "${repo}".`);
```

Comment:
This can be removed. Let the caller of this method handle errors and show message if necessary.